### ISSUE TITLE
RocketClassifier's fit_predict_proba and predict_proba methods don't do anything useful

### ISSUE THREAD

**Issue by collindbell** [sktime/sktime#3626]
https://github.com/sktime/sktime/blob/4194aacaa105ef59807ea56c1bb3497b37eb1613/sktime/classification/base.py#L384-L425

The fit_predict_proba/predict_proba methods don't appear to do anything worthwhile because RidgeClassifierCV doesn't have a predict_proba method itself and isn't really intended to output probabilistic values. Currently, the fit_predict_proba and predict_proba methods for RocketClassifier just output a numpy array for each member of the sample with a 1 for the decided class and a 0 for all other classes.

Recommend removal of these methods or approximating probabilities with softmax as discussed here: https://stackoverflow.com/questions/22538080/scikit-learn-ridge-classifier-extracting-class-probabilities

**Comment by fkiraly** [comment]
This behaviour is the "default" behaviour for the proba methods if no smarter behaviour is implemented.

It is better to have a reasonable default than not have those methods at all.

So, we won't remove them as it would break the interface contract (all classifiers have this implemented).

But, if you think there is something smarter that would make sense, happy to see the suggestion or a PR.

**Comment by TonyBagnall** [comment]
hi, thanks for this. As Franz says, this is the default behaviour, and if there is no default for predict_proba other code expecting classifiers to have this function will break. This RocketClassifier is meant to statically capture the implementation described here https://link.springer.com/article/10.1007/s10618-020-00701-z. If you want sensible probabilities, you can make your own rocket based classifier using a pipeline and a scikit learn classifier that does this.

**Comment by fkiraly** [comment]
closing since default behaviour and follows the interface contract, agree with @TonyBagnall
